packages/xchain-ethereum/src/client.ts

```diff
@@ -227,64 +270,168 @@ export default class Client implements EthereumClient {
     return this.provider.getTransactionCount(address || this.getAddress(), blocktag)
   }
 
-  /**
-   * Gets the transaction history of an address.
-   */
-  async getTransactions(address: Address = this._address): Promise<Array<TransactionResponse>> {
-    if (address && !Client.validateAddress(address)) {
-      return Promise.reject('Invalid Address')
-    } else {
-      const transactions = this._etherscan.getHistory(address)
-      return transactions
+  getTransactions = async (params?: TxHistoryParams): Promise<TxsPage> => {
+    const address = params?.address ?? this.getAddress()
+    const limit = params?.limit ?? 10
+    const offset = params?.offset ?? 0
+
+    let totalCount = 0
+    const transactions: Tx[] = []
+    try {
+      //Calling getAddress without limit/offset to get total count
+      const dAddr = await blockChair.getAddress(this.blockchairNodeUrl, address, this.blockchairNodeApiKey)
+      totalCount = dAddr[address].transactions.length
+
+      const dashboardAddress = await blockChair.getAddress(
+        this.blockchairNodeUrl,
+        address,
+        this.blockchairNodeApiKey,
+        limit,
+        offset,
+      )
+      const txList = dashboardAddress[address].transactions
+
+      for (const hash of txList) {
+        const rawTx = (await blockChair.getTx(this.blockchairNodeUrl, hash, this.blockchairNodeApiKey))[hash]
+        const tx: Tx = {
+          asset: AssetETH,
+          from: rawTx.inputs.map((i: TxIO) => ({ from: i.recipient, amount: baseAmount(i.value, ETH_DECIMAL) })),
+          to: rawTx.outputs
+            // ignore tx with type 'nulldata'
+            .filter((i: TxIO) => i.type !== 'nulldata')
+            .map((i: TxIO) => ({ to: i.recipient, amount: baseAmount(i.value, ETH_DECIMAL) })),
```

Comment:
Do we really know that decimal is `ETH_DECIMAL`? What about ERC20 token, which might have different decimal?